src/terminal.rs

```diff
@@ -162,6 +210,10 @@ where
         self.buffers[1 - self.current].reset();
         self.buffers[1 - self.current].resize(area);
         self.known_size = area;
+        if let CursorState::Shown { x, y } = self.cursor_state_after_drawing {
+            let (x, y) = clip(self.known_size, (x, y));
```

Comment:
the crate usually does not attempt to fix silently users' errors. In this case if the cursor is outside of the viewport then it should simply not be shown.

Reply:
Should I explicitly check for that and hide the cursor, or is it okay to just pass the coordinates down to `Terminal::set_cursor`, and let the backend deal with it?

Reply:
I went ahead and changed it so it just passes the coordinates down into `Terminal::set_cursor()`; see https://github.com/fdehau/tui-rs/pull/309/commits/65db02ab339fd20f8187eb9ac13157bc655321b7.